live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/config/properties/Transform.java

```diff
@@ -27,16 +27,28 @@
 
 import static java.util.Objects.requireNonNullElseGet;
 
+import java.util.Collections;
 import java.util.Map;
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
 import jakarta.validation.constraints.Positive;
 
 import org.springframework.boot.context.properties.NestedConfigurationProperty;
+import org.springframework.boot.context.properties.bind.ConstructorBinding;
 import org.springframework.boot.context.properties.bind.DefaultValue;
 
-public record Transform(@NotNull Request request, @NotNull Response response, @NotNull SharedFileStore sharedFileStore, @NotNull MimeType mimeType)
+public record Transform(@NotNull Request request, @NotNull Response response, @NotNull SharedFileStore sharedFileStore, MimeType mimeType)
 {
+
+    @ConstructorBinding
+    public Transform(@NotNull Request request, @NotNull Response response, @NotNull SharedFileStore sharedFileStore, MimeType mimeType)
+    {
+        this.request = request;
+        this.response = response;
+        this.sharedFileStore = sharedFileStore;
+        this.mimeType = mimeType != null ? mimeType : new MimeType(Collections.emptyMap());
```

Comment:
I wonder if this constructor could be replaced with:
```
public Transform
{
    mimeType = requireNonNullElse(mimeType, new MimeType(Collections.emptyMap()));
}
```
🤔

Reply:
Thanks for the hint - it works.